File: result/translate/chatgpt/python2java/java/REMOVE_BRACKETS_ALGEBRAIC_STRING_CONTAINING_OPERATORS.java
public static String f_gold(String Str) {
    int Len = Str.length();
    char[] res = new char[Len];
    int index = 0;
    int i = 0;
    Stack<Integer> s = new Stack<Integer>();
    s.push(0);
    while (i < Len) {
        if (Str.charAt(i) == '+') {
            if (s.peek() == 1) {
                res[index] = '-';
                index++;
            }
            if (s.peek() == 0) {
                res[index] = '+';
                index++;
            }
        } else if (Str.charAt(i) == '-') {
            if (s.peek() == 1) {
                res[index] = '+';
                index++;
            } else if (s.peek() == 0) {
                res[index] = '-';
                index++;
            }
        } else if (Str.charAt(i) == '(' && i > 0) {
            if (Str.charAt(i - 1) == '-') {
                int x = (s.peek() == 1) ? 0 : 1;
                s.push(x);
            } else if (Str.charAt(i - 1) == '+') {
                s.push(s.peek());
            }
        } else if (Str.charAt(i) == ')') {
            s.pop();
        } else {
            res[index] = Str.charAt(i);
            index++;
        }
        i++;
    }
    return new String(res);
}

File: result/translate/chatgpt/java2python/python/REMOVE_BRACKETS_ALGEBRAIC_STRING_CONTAINING_OPERATORS.py
def f_gold(str):
    len_str = len(str)
    res = [''] * len_str
    index = 0
    i = 0
    s = []
    s.append(0)
    
    while i < len_str:
        if str[i] == '+':
            if s[-1] == 1:
                res[index] = '-'
                index += 1
            if s[-1] == 0:
                res[index] = '+'
                index += 1
        elif str[i] == '-':
            if s[-1] == 1:
                res[index] = '+'
                index += 1
            elif s[-1] == 0:
                res[index] = '-'
                index += 1
        elif str[i] == '(' and i > 0:
            if str[i - 1] == '-':
                x = 0 if s[-1] == 1 else 1
                s.append(x)
            elif str[i - 1] == '+':
                s.append(s[-1])
        elif str[i] == ')':
            s.pop()
        else:
            res[index] = str[i]
            index += 1
        i += 1
        
    return ''.join(res)